Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -273,13 +278,107 @@ void HoI4FocusTree::addGenericFocusTree(const set<string>& majorIdeologies)
 		throw std::runtime_error("Could not load focus technology_sharing");
 	}
 
+	if (const auto& originalFocus = loadedFocuses.find("border_disputes"); originalFocus != loadedFocuses.end())
+	{
+		auto newFocus = make_shared<HoI4::SharedFocus>(originalFocus->second);
+		if (majorIdeologies.contains("communism"))
+		{
+			std::string comm;
+			comm = "if = {\n";
+			comm += "\t\t\t\t\t\tlimit = { ROOT = { has_government = communism } }\n";
+			comm += "\t\t\t\t\t\tpotential_communist_puppet_target = yes\n";
+			comm += "\t\t\t\t\t}";
+			newFocus->updateFocusElement(newFocus->available, "#COMMUNISM", comm);
+			newFocus->updateFocusElement(newFocus->selectEffect, "#COMMUNISM", "potential_communist_puppet_target = yes");
+		}
+		else
+		{
+			newFocus->removePlaceholder(newFocus->available, "#COMMUNISM");
+			newFocus->removePlaceholder(newFocus->selectEffect, "#COMMUNISM");
+		}
+		if (majorIdeologies.contains("fascism"))
+		{
+			std::string fasc;
+			fasc = "if = {\n";
+			fasc += "\t\t\t\t\t\tlimit = { ROOT = { has_government = fascism } }\n";
+			fasc += "\t\t\t\t\t\tpotential_fascist_annex_target = yes\n";
+			fasc += "\t\t\t\t\t}";
+			newFocus->updateFocusElement(newFocus->available, "#FASCISM", fasc);
+			newFocus->updateFocusElement(newFocus->selectEffect, "#FASCISM", "potential_fascist_annex_target = yes");
+		}
+		else
+		{
+			newFocus->removePlaceholder(newFocus->available, "#FASCISM");
+			newFocus->removePlaceholder(newFocus->selectEffect, "#FASCISM");
+		}
+		sharedFocuses.push_back(newFocus);
+	}
+	else
```

Comment:
Are you sure every country you're touching has border disputes branch?

Reply:
> Are you sure every country you're touching has border disputes branch?

These are generic focus trees, every country gets them. Just like the technology_sharing above the ones I added.